fdbserver/RESTKmsConnector.actor.cpp

```diff
@@ -367,30 +396,141 @@ void parseKmsResponse(Reference<RESTKmsConnectorCtx> ctx,
 
 		// Extract cipher refresh and/or expiry interval if supplied
-		Optional<int64_t> refreshAfterSec = cipherDetail.HasMember(CIPHER_KEY_REFRESH_AFTER_SEC) &&
-		                                            cipherDetail[CIPHER_KEY_REFRESH_AFTER_SEC].GetInt64() > 0
-		                                        ? cipherDetail[CIPHER_KEY_REFRESH_AFTER_SEC].GetInt64()
-		                                        : Optional<int64_t>();
-		Optional<int64_t> expireAfterSec = cipherDetail.HasMember(CIPHER_KEY_EXPIRE_AFTER_SEC)
-		                                       ? cipherDetail[CIPHER_KEY_EXPIRE_AFTER_SEC].GetInt64()
-		                                       : Optional<int64_t>();
-
-		outCipherKeyDetails->emplace_back_deep(*arena,
-		                                       cipherDetail[ENCRYPT_DOMAIN_ID_TAG].GetInt64(),
-		                                       cipherDetail[BASE_CIPHER_ID_TAG].GetUint64(),
-		                                       StringRef(cipherKey.get(), cipherKeyLen),
-		                                       refreshAfterSec,
-		                                       expireAfterSec);
+		Optional<int64_t> refreshAfterSec =
+		    cipherDetail.HasMember(REFRESH_AFTER_SEC) && cipherDetail[REFRESH_AFTER_SEC].GetInt64() > 0
+		        ? cipherDetail[REFRESH_AFTER_SEC].GetInt64()
+		        : Optional<int64_t>();
+		Optional<int64_t> expireAfterSec =
+		    cipherDetail.HasMember(EXPIRE_AFTER_SEC) ? cipherDetail[EXPIRE_AFTER_SEC].GetInt64() : Optional<int64_t>();
+
+		result.emplace_back_deep(result.arena(),
+		                         cipherDetail[ENCRYPT_DOMAIN_ID_TAG].GetInt64(),
+		                         cipherDetail[BASE_CIPHER_ID_TAG].GetUint64(),
+		                         StringRef(cipherKey.get(), cipherKeyLen),
+		                         refreshAfterSec,
+		                         expireAfterSec);
 	}
 
-	if (doc.HasMember(KMS_URLS_TAG)) {
-		try {
-			extractKmsUrls(ctx, doc, resp);
-		} catch (Error& e) {
-			TraceEvent("RefreshKmsUrls_Failed", ctx->uid).error(e);
-			// Given cipherKeyDetails extraction was done successfully, ignore KmsUrls parsing error
+	checkDocForNewKmsUrls(ctx, resp, doc);
+
+	return result;
+}
+
+Standalone<VectorRef<BlobMetadataDetailsRef>> parseBlobMetadataResponse(Reference<RESTKmsConnectorCtx> ctx,
+                                                                        Reference<HTTP::Response> resp) {
+	// Acceptable response payload json format:
+	// (baseLocation and partitions follow the same properties as described in BlobMetadataUtils.h)
+	//
+	// response_json_payload {
+	//   "blob_metadata_details" : [
+	//     {
+	//        "domain_id" : <domainId>,
+	//        "domain_name" : <baseCipher>,
+	//        "baseLocation" : <baseLocation>, (Optional if partitions is present)
+	//        "partitions" : [
+	//			  "partition1", "partition2", ...
+	//		  ], (Optional if baseLocation is present)
+	//        "refresh_after_sec"   : <refreshCipherTimeInterval>, (Optional)
```

Comment:
nit: `refreshCipherTimeInterval`  -> `refresh`TimeInterval` here and for EncryptCipher method as well I think